### DISCUSSION TITLE
SetValue during schema validation

### DISCUSSION THREAD

**Discussion by daniloak** [colinhacks/zod#2922] [Q&A]
Is it possible to set another value during the schema validation? 
In this situation, let's say I have another property related to the user, could I set its value after verifying its existence? 

```ts
const userId = z.string().refine(async (id) => {
  // verify that ID exists in database
  return true;
});
```

**Comment by JacobWeisenburger** [comment]
is this what you are looking for?

https://github.com/colinhacks/zod#parseasync

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
